lib/ansible/executor/task_executor.py

```diff
@@ -1010,13 +1010,17 @@ def _get_action_handler(self, connection, templar):
 
         module_prefix = self._task.action.split('_')[0]
 
+        collections = self._task.collections
+
         # let action plugin override module, fallback to 'normal' action plugin otherwise
-        if self._task.action in self._shared_loader_obj.action_loader:
```

Comment:
It is kind of tempting to remove the PluginLoader's `__contains__ = has_plugin`, to avoid accidentally using
the non collection_list aware version of PluginLoader.has_plugin.

I've seen a lot of hard to debug behavior that was rooted in code that does `if some_plugin in some_loader`. Removing
the `__contains__` should expose where it is used so it can be fix/removed.